solr/solrj/src/java/org/apache/solr/common/util/TextWriter.java

```diff
@@ -88,7 +87,7 @@ default void writeVal(String name, Object val, boolean raw) throws IOException {
     } else if (val instanceof MapWriter) {
       writeMap(name, (MapWriter) val);
     } else if (val instanceof ReflectWritable) {
-      writeMap(name, new DelegateMapWriter(val));
+      writeVal(name, Utils.getReflectWritable(val));
```

Comment:
Ok, the name is probably really bad here, but `getReflectWritable()` does not return a `ReflectWritable` class. It returns that `DelegateReflectWriter`. I'll rename this to `getReflectWriter()`, as it will be much less confusing. Good catch.